### PR TITLE
EPMDJ-9249: Decompose compound cell

### PR DESCRIPTION
**Pull request by dmitrii-kliushentsev**

### CODE REVIEW

File: src/components/forms/inputs/button/copy-button/copy-button.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import 'twin.macro';
+import { Icons } from '../../../../icon';
```
Comment (IliaSolovev) [inline src/components/forms/inputs/button/copy-button/copy-button.tsx:18]: Alias?

File: src/components/forms/inputs/button/copy-button/copy-button.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { useCopy } from '../../../../../hooks';
+import { copyToClipboard } from '../../../../../utils';
+
+interface Props {
+  text: string,
+}
```
Comment (IliaSolovev) [inline src/components/forms/inputs/button/copy-button/copy-button.tsx:24]:
```suggestion
}

```

File: src/components/forms/inputs/button/copy-button/copy-button.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+export const CopyButton = ({ text }: Props) => {
```
Comment (IliaSolovev) [inline src/components/forms/inputs/button/copy-button/copy-button.tsx:25]: As i remember we decide to use children here

File: src/components/table/cellcomponents/name.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import tw, { styled } from 'twin.macro';
+
+export const Name = styled.div`
+  ${tw`text-ellipsis h-5 leading-20`}
+
+  ${({ bold }: { bold: boolean }) => bold && tw`font-bold`}
```
Comment (IliaSolovev) [inline src/components/table/cellcomponents/name.tsx:6]: we can set it inline without props

File: src/components/table/cellcomponents/link.tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React, { FC } from 'react';
+
+export const Link: FC = ({ children }) => (
```
Comment (IliaSolovev) [inline src/components/table/cellcomponents/link.tsx:3]: We already have link styles

File: src/components/table/cellcomponents/additional-info.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+interface Props {
+  info: string,
+}
+
+export const AdditionalInfo = ({ info }: Props) => (
+  <div
+    tw="text-ellipsis mt-1 h-5 leading-20 text-12 font-regular text-monochrome-default"
+    data-test="compound-cell:additional-info"
```
Comment (IliaSolovev) [inline src/components/table/cellcomponents/additional-info.tsx:10]: Why here you write compound cell if it is not a compound cell component?

File: src/components/table/cellcomponents/index.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { Name } from './name';
+import { Link } from './link';
+import { AdditionalInfo } from './additional-info';
+
+export const CellComponents = {
```
Comment (IliaSolovev) [inline src/components/table/cellcomponents/index.ts:5]: I know that this is a component, why note about it again

File: src/components/table/cellcomponents/index.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+  Link,
```
Comment (IliaSolovev) [inline src/components/table/cellcomponents/index.ts:6]: as I understand it all this is part of one component? so why are you creating different files for it?

File: src/components/table/cells/compound-cell/compound-cell.tsx [hunk 9]
```diff
@@ -13,69 +13,21 @@
...
-          >
-            {cellAdditionalInfo}
-          </div>
-        )}
-      </div>
+  icon, children, ...rest
+}: Props) => (
+  <div tw="flex gap-2 py-2 text-monochrome-black" {...rest}>
+    <div tw="h-5 flex items-center">{icon}</div>
+    <div className="text-ellipsis group">
```
Comment (IliaSolovev) [inline src/components/table/cells/compound-cell/compound-cell.tsx:29]: className?

**Review by IliaSolovev — CHANGES_REQUESTED** [review]

**Review by RomanAchilov — CHANGES_REQUESTED** [review]
It is necessary to synchronize the cells with the design system. Make basic components -> make complex ones inside the main application and using tailwind styling. Also prepare hooks for complex logic.

File: src/components/table/cells/compound-cell/compound-cell.tsx [hunk 10]
```diff
@@ -30,52 +26,38 @@ interface Props {
...
-            {cellAdditionalInfo}
+}: Props) => (
+  <div tw="flex gap-2 py-2 text-monochrome-black" {...rest}>
+    <div tw="h-5 flex items-center">{icon}</div>
+    <div tw="text-ellipsis group">
+      <div tw="flex gap-x-2 items-center">
+        <Name
+          data-test="compound-cell:name"
+          title={cellName}
+          bold={!!cellAdditionalInfo}
```
Comment (IliaSolovev) [inline src/components/table/cells/compound-cell/compound-cell.tsx:37]:
```suggestion
          bold={Boolean(cellAdditionalInfo)}
```

**Review by IliaSolovev — APPROVED** [review]

**Review by IliaSolovev — APPROVED** [review]

**Review by IliaSolovev — APPROVED** [review]

**Pull request merged**